### PR TITLE
[AIR] Don't ravel predictions in `TensorflowPrediction.predict`

### LINKED ISSUE #25137: [AIR] `TensorflowPredictor` incorrectly ravels predictions
### What happened + What you expected to happen

I used `BatchPredictor` to classify an image dataset. I expected the output dataset to contain logits, but instead it contains scalars.

I got 
```
>>> outputs.show(1)
{'predictions': 0.0}
```

but I should've got

```
{'predictions': array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32)}
```

**Importat: you'll only run into this error if you've patched #25125 and #25124 first!**

### Versions / Dependencies

**Ray**: master
**OS**: MacOS
**Python**: 3.8.12

### Reproduction script

```
import ray
from ray.ml.predictors.integrations.tensorflow import TensorflowPredictor
import tensorflow as tf
from tensorflow.keras import models, layers
from ray.ml.batch_predictor import BatchPredictor
from ray.ml.checkpoint import Checkpoint

batch_size = 4
height = 32
width = 32
num_channels = 3
num_classes = 10


def build_model():
    model = models.Sequential()

    model.add(
        layers.Conv2D(
            6, (5, 5), activation="relu", input_shape=(height, width, num_channels)
        )
    )
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Conv2D(16, (5, 5), activation="relu"))
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Flatten())
    model.add(layers.Dense(120, activation="relu"))
    model.add(layers.Dense(84, activation="relu"))
    model.add(layers.Dense(num_classes))
    return model


# Mocking a checkpoint so I can use `BatchPredictor`.
model = build_model()
model.build(input_shape=(0, 32, 32, 3))
checkpoint = Checkpoint.from_dict({"model": model.get_weights()})

batch_predictor = BatchPredictor.from_checkpoint(
    checkpoint=checkpoint,
    predictor_cls=TensorflowPredictor,
    model_definition=build_model,
)

dataset = ray.data.range_tensor(batch_size, shape=(height, width, num_channels))
outputs = batch_predictor.predict(dataset)
outputs.show(1)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

`TensorflowPredictor.predict` doesn't correctly produce logits. For more information, see #25137.

## Related issue number

Fixes #25137 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

File: python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py [hunk 1]
```diff
@@ -147,5 +147,5 @@ def build_model(self):
         if self.model_weights:
             model.set_weights(self.model_weights)
 
-        prediction = model(tensor).numpy().ravel()
-        return pd.DataFrame(prediction, columns=["predictions"])
+        prediction = list(model(tensor).numpy())
```
Comment (xwjiang2010) [inline python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:150]: cc @JiahaoYao 

hmmm, This doesn't work for general case, right? For example, Jimmy is outputting an image per prediction.
I thought yesterday I saw you using TensorArray?

Comment (bveeramani) [reply python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:150]: I was yesterday, but I thought we decided to use a list so there isn't a dependency on Ray Data?

In any case, I'd prefer to use `TensorArray`. I feel like it works a bit better.

Comment (bveeramani) [reply python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:150]: Another issue with the `list` approach is that we can't show outputs from `BatchPredictor.predict`.

```
outputs = batch_predictor.predict(
    data=test_dataset, feature_columns=["image"], unsqueeze=False
) 
outputs.show(1)  # Raises error during `repr`!
```

Comment (xwjiang2010) [reply python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:150]: @amogkam Are you ok with using TensorArray here and leave a TODO to factor out TensorArray to be just a util (independent of Ray Data)

Comment (xwjiang2010) [reply python/ray/ml/predictors/integrations/tensorflow/tensorflow_predictor.py:150]: synced offline. Will go with list() for now and revisit after Clark is back.

**Review by xwjiang2010 — APPROVED** [review]

**Review by amogkam — APPROVED** [review]
